awx/main/scheduler/task_manager.py

```diff
@@ -509,50 +442,91 @@ def process_pending_tasks(self, pending_tasks):
                     running_workflow_templates.add(task.unified_job_template_id)
                 self.start_task(task, None, task.get_jobs_fail_chain(), None)
                 continue
+            # by popular request, e.g. to make Alan happy, we will ensure we have a control node before we look for execution
+            # capacity
+            controller_node = self.task_manager_instances.best_control_node_candidate()
+            task.controller_node = controller_node
+            if not controller_node:
+                found_acceptable_queue = False
+                # There is not enough control capacity, we can't run any more tasks. Stop processing pending tasks.
+                logger.warning("No control plane nodes available, will stop processing pending tasks")
+                # TODO: if we break here, we don't get to do
+                # task.log_lifecycle("needs_capacity")
+                # for all the tasks
+                break
+
+            if isinstance(task, ProjectUpdate):
+                controller_node = self.task_manager_instances.assign_task_to_control_node(task, self.ig_capacity_graph, controller_node)
```

Comment:
You already picked this, I don't see any need to pick it again. Can't you just delete this line?